SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx

```diff
@@ -262,7 +267,7 @@ int vtkSlicerSkeletalRepresentationInitializerLogic::SetInputFileName(const std:
     // save
     std::string tempDir = this->GetApplicationLogic()->GetTemporaryPath();
     char tempFileName[MAX_FILE_NAME];
-    sprintf(tempFileName, "%s/temp_output.vtk", tempDir.c_str());
+    sprintf(tempFileName, "%s/initial_surface.vtk", tempDir.c_str());
```

Comment:
use `std::string` instead of `char` and use `std::string` operations to joing strings instead of the c-style `sprintf`.


Hitting `char` and `sprintf` related errors at build time:


Dumping errors:
```
SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx: In member function ‘int vtkSlicerSkeletalRepresentationInitializerLogic::GenerateSrepForEllipsoid(vtkPolyData*, int, int, int)’:
SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:934:25: warning: ‘/up’ directive writing 3 bytes into a region of size between 1 and 256 [-Wformat-overflow=]
     sprintf(upFileName, "%s/up%04d.vtk", modelFolder, totalNum);
                         ^~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:873,
                 from /usr/include/c++/8.2.1/cstdio:42,
                 from /usr/include/c++/8.2.1/ext/string_conversions.h:43,
                 from /usr/include/c++/8.2.1/bits/basic_string.h:6391,
                 from /usr/include/c++/8.2.1/string:52,
                 from /usr/include/c++/8.2.1/bits/locale_classes.h:40,
                 from /usr/include/c++/8.2.1/bits/ios_base.h:41,
                 from /usr/include/c++/8.2.1/ios:42,
                 from /usr/include/c++/8.2.1/ostream:38,
                 from /usr/include/c++/8.2.1/iostream:39,
                 from VTK/Common/Core/vtkIOStream.h:31,
                 from VTK/Common/Core/vtkSystemIncludes.h:39,
                 from VTK/Common/Core/vtkIndent.h:28,
                 from VTK/Common/Core/vtkObjectBase.h:52,
                 from VTK/Common/Core/vtkObject.h:45,
                 from slicersources-src/Libs/MRML/Core/vtkObserverManager.h:22,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLAbstractLogic.h:17,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerModuleLogic.h:18,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.h:28,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:19:
/usr/include/bits/stdio2.h:36:34: note: ‘__builtin___sprintf_chk’ output between 12 and 274 bytes into a destination of size 256
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:950:27: warning: ‘/down’ directive writing 5 bytes into a region of size between 1 and 256 [-Wformat-overflow=]
     sprintf(downFileName, "%s/down%04d.vtk", modelFolder, totalNum);
                           ^~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:873,
                 from /usr/include/c++/8.2.1/cstdio:42,
                 from /usr/include/c++/8.2.1/ext/string_conversions.h:43,
                 from /usr/include/c++/8.2.1/bits/basic_string.h:6391,
                 from /usr/include/c++/8.2.1/string:52,
                 from /usr/include/c++/8.2.1/bits/locale_classes.h:40,
                 from /usr/include/c++/8.2.1/bits/ios_base.h:41,
                 from /usr/include/c++/8.2.1/ios:42,
                 from /usr/include/c++/8.2.1/ostream:38,
                 from /usr/include/c++/8.2.1/iostream:39,
                 from VTK/Common/Core/vtkIOStream.h:31,
                 from VTK/Common/Core/vtkSystemIncludes.h:39,
                 from VTK/Common/Core/vtkIndent.h:28,
                 from VTK/Common/Core/vtkObjectBase.h:52,
                 from VTK/Common/Core/vtkObject.h:45,
                 from slicersources-src/Libs/MRML/Core/vtkObserverManager.h:22,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLAbstractLogic.h:17,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerModuleLogic.h:18,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.h:28,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:19:
/usr/include/bits/stdio2.h:36:34: note: ‘__builtin___sprintf_chk’ output between 14 and 276 bytes into a destination of size 256
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:981:27: warning: ‘/mesh’ directive writing 5 bytes into a region of size between 1 and 256 [-Wformat-overflow=]
     sprintf(meshFileName, "%s/mesh%04d.vtk", modelFolder, totalNum);
                           ^~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:873,
                 from /usr/include/c++/8.2.1/cstdio:42,
                 from /usr/include/c++/8.2.1/ext/string_conversions.h:43,
                 from /usr/include/c++/8.2.1/bits/basic_string.h:6391,
                 from /usr/include/c++/8.2.1/string:52,
                 from /usr/include/c++/8.2.1/bits/locale_classes.h:40,
                 from /usr/include/c++/8.2.1/bits/ios_base.h:41,
                 from /usr/include/c++/8.2.1/ios:42,
                 from /usr/include/c++/8.2.1/ostream:38,
                 from /usr/include/c++/8.2.1/iostream:39,
                 from VTK/Common/Core/vtkIOStream.h:31,
                 from VTK/Common/Core/vtkSystemIncludes.h:39,
                 from VTK/Common/Core/vtkIndent.h:28,
                 from VTK/Common/Core/vtkObjectBase.h:52,
                 from VTK/Common/Core/vtkObject.h:45,
                 from slicersources-src/Libs/MRML/Core/vtkObserverManager.h:22,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLAbstractLogic.h:17,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerModuleLogic.h:18,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.h:28,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:19:
/usr/include/bits/stdio2.h:36:34: note: ‘__builtin___sprintf_chk’ output between 14 and 276 bytes into a destination of size 256
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:1033:28: warning: ‘/crest’ directive writing 6 bytes into a region of size between 1 and 256 [-Wformat-overflow=]
     sprintf(crestFileName, "%s/crest%04d.vtk", modelFolder, totalNum);
                            ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:873,
                 from /usr/include/c++/8.2.1/cstdio:42,
                 from /usr/include/c++/8.2.1/ext/string_conversions.h:43,
                 from /usr/include/c++/8.2.1/bits/basic_string.h:6391,
                 from /usr/include/c++/8.2.1/string:52,
                 from /usr/include/c++/8.2.1/bits/locale_classes.h:40,
                 from /usr/include/c++/8.2.1/bits/ios_base.h:41,
                 from /usr/include/c++/8.2.1/ios:42,
                 from /usr/include/c++/8.2.1/ostream:38,
                 from /usr/include/c++/8.2.1/iostream:39,
                 from VTK/Common/Core/vtkIOStream.h:31,
                 from VTK/Common/Core/vtkSystemIncludes.h:39,
                 from VTK/Common/Core/vtkIndent.h:28,
                 from VTK/Common/Core/vtkObjectBase.h:52,
                 from VTK/Common/Core/vtkObject.h:45,
                 from slicersources-src/Libs/MRML/Core/vtkObserverManager.h:22,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLAbstractLogic.h:17,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerModuleLogic.h:18,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.h:28,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:19:
/usr/include/bits/stdio2.h:36:34: note: ‘__builtin___sprintf_chk’ output between 15 and 277 bytes into a destination of size 256
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:1117:28: warning: ‘/curve’ directive writing 6 bytes into a region of size between 1 and 256 [-Wformat-overflow=]
     sprintf(curveFileName, "%s/curve%04d.vtk", modelFolder, totalNum);
                            ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:873,
                 from /usr/include/c++/8.2.1/cstdio:42,
                 from /usr/include/c++/8.2.1/ext/string_conversions.h:43,
                 from /usr/include/c++/8.2.1/bits/basic_string.h:6391,
                 from /usr/include/c++/8.2.1/string:52,
                 from /usr/include/c++/8.2.1/bits/locale_classes.h:40,
                 from /usr/include/c++/8.2.1/bits/ios_base.h:41,
                 from /usr/include/c++/8.2.1/ios:42,
                 from /usr/include/c++/8.2.1/ostream:38,
                 from /usr/include/c++/8.2.1/iostream:39,
                 from VTK/Common/Core/vtkIOStream.h:31,
                 from VTK/Common/Core/vtkSystemIncludes.h:39,
                 from VTK/Common/Core/vtkIndent.h:28,
                 from VTK/Common/Core/vtkObjectBase.h:52,
                 from VTK/Common/Core/vtkObject.h:45,
                 from slicersources-src/Libs/MRML/Core/vtkObserverManager.h:22,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLAbstractLogic.h:17,
                 from slicersources-src/Libs/MRML/Logic/vtkMRMLApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerApplicationLogic.h:25,
                 from slicersources-src/Base/Logic/vtkSlicerModuleLogic.h:18,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.h:28,
                 from SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:19:
/usr/include/bits/stdio2.h:36:34: note: ‘__builtin___sprintf_chk’ output between 15 and 277 bytes into a destination of size 256
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx: In member function ‘int vtkSlicerSkeletalRepresentationInitializerLogic::FlowSurfaceMesh(const string&, double, double, int, int)’:
SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx:410:27: warning: ‘%04d’ directive writing between 4 and 10 bytes into a region of size between 0 and 255 [-Wformat-overflow=]
         sprintf(fileName, "%s/%04d.vtk", forwardFolder, iter+1);
```